File: scripts/process_train_results.R
#Author:Devan Rehunathan
#Date:August 2009
#Name:process_train_results.R

# Constants
number_of_weeks <- 51;
number_of_days <- 6;
number_of_hours <- 17;
number_of_stations <- 26;
number_of_trains <- 6;

mean <- 0;
standard_deviation <- 0;

    for(w in 0:number_of_weeks){
        for(d in 0:number_of_days){
            for(h in 0:number_of_hours){
                for(s in 0:number_of_stations){

                    current_file <- paste("/home/dr/RailwaySimulator/results/train_results", "/Week", w, "/Day",  d,  "/Hour",  h,  "/Station",  s, ".txt", sep="");

                    input_table <-read.table(current_file,header=FALSE,sep="\t");
                    
                        if( (w+d+h+s)==0){
                            train_1 <- input_table[(1), ];
                            train_2 <- input_table[(2), ];
                            train_3 <- input_table[(3), ];
                            train_4 <- input_table[(4), ];
                            train_5 <- input_table[(5), ];
                            train_6 <- input_table[(6), ];
                            train_7 <- input_table[(7), ];
                        }
                        else{
                            train_1 <-rbind(train_1, input_table[(1), ]);
                            train_2 <-rbind(train_2, input_table[(2), ]);
                            train_3 <-rbind(train_3, input_table[(3), ]);
                            train_4 <-rbind(train_4, input_table[(4), ]);
                            train_5 <-rbind(train_5, input_table[(5), ]);
                            train_6 <-rbind(train_6, input_table[(6), ]);
                            train_7 <-rbind(train_7, input_table[(7), ]);
                        }
                    }
                }         
            }
        }

write.table(train_1, file="/home/dr/RailwaySimulator/results/train_1.txt", row.name=F, col.names=F,  sep="\t");
write.table(train_2, file="/home/dr/RailwaySimulator/results/train_2.txt", row.name=F, col.names=F,  sep="\t");
write.table(train_3, file="/home/dr/RailwaySimulator/results/train_3.txt", row.name=F, col.names=F,  sep="\t");
write.table(train_4, file="/home/dr/RailwaySimulator/results/train_4.txt", row.name=F, col.names=F,  sep="\t");
write.table(train_5, file="/home/dr/RailwaySimulator/results/train_5.txt", row.name=F, col.names=F,  sep="\t");
write.table(train_6, file="/home/dr/RailwaySimulator/results/train_6.txt", row.name=F, col.names=F,  sep="\t");
write.table(train_7, file="/home/dr/RailwaySimulator/results/train_7.txt", row.name=F, col.names=F,  sep="\t");

q()

File: scripts/process_train_results.py
import string, re, sys

weeks = 52
days = 7
hours = 18
stations = 27
trains = 7

results_folder = sys.argv[1];

lnum = 0

o0 = open(results_folder + '/train_0.dat', 'a')
o1 = open(results_folder + '/train_1.dat', 'a')
o2 = open(results_folder + '/train_2.dat', 'a')
o3 = open(results_folder + '/train_3.dat', 'a')
o4 = open(results_folder + '/train_4.dat', 'a')
o5 = open(results_folder + '/train_5.dat', 'a')
o6 = open(results_folder + '/train_6.dat', 'a')

# Go through all StationX.txt files
for w in range(0, weeks):
	for d in range(0, days):
		for h in range(0, hours):
			for s in range(0, stations):

				current_file = results_folder + '/train_results/Week' + str(w) + '/Day' + str(d) + '/Hour' + str(h) + '/Station' + str(s) + '.txt'  
				f = open(current_file, 'r')

				for line in f:
                                        
					#Counter for line number. Each file has 7 lines
					lnum += 1

                                        #Split text file into columns
					columns = line.split("\t")
					
                                        #We are interested only in 2nd=registrations and 5th=handovers
					output_line = str(columns[2]) + '\t' + str(columns[5]) 

                                        #Split handover and registrations into separate train identities
					if(lnum%7 == 1):
						o0.write(output_line)
					if(lnum%7 == 2):
						o1.write(output_line)
					if(lnum%7 == 3):
						o2.write(output_line)
					if(lnum%7 == 4):
						o3.write(output_line)
					if(lnum%7 == 5):
						o4.write(output_line)
					if(lnum%7 == 6):
						o5.write(output_line)
					if(lnum%7 == 0):
						o6.write(output_line)

